lib/devise/multi_email/parent_model_manager.rb

```
@@ -4,51 +4,86 @@ module Devise
 
-      def initialize(record)
-        @record = record
+      def current_email_record
+        login_email_record || primary_email_record
       end
 
-      # Gets the email records that have not been deleted
-      def filtered_emails
-        emails.reject(&:destroyed?).reject(&:marked_for_destruction?)
+      def login_email_record
+        if @parent_record.current_login_email.present?
+          formatted_email = format_email(@parent_record.current_login_email)
+          filtered_emails.find{ |item| item.email == formatted_email }
+        end
       end
 
       # Gets the primary email record.
-      def primary_email
+      def primary_email_record
         filtered_emails.find(&:primary?)
       end
-      alias_method Devise::MultiEmail.primary_email_method_name, :primary_email
-
-      def change_primary_email_to(new_email)
-        # Use Devise formatting settings for emails
-        formatted_email = record.class.send(:devise_parameter_filter).filter(email: new_email)[:email]
-
-        valid_emails = filtered_emails
+      alias_method Devise::MultiEmail.primary_email_method_name, :primary_email_record
 
+      # :force_primary option sets this email record to primary
+      # :skip_confirmations option confirms this email record (without saving)
+      # @see `set_primary_record_to`
+      def change_primary_email_to(new_email, options = {})
         # mark none as primary when set to nil
         if new_email.nil?
-          valid_emails.each{ |record| record.primary = false }
+          filtered_emails.each{ |item| item.primary = false }
 
         # select or build an email record
         else
-          record = valid_emails.find{ |record| record.email == formatted_email }
+          record = find_or_build_for_email(new_email)
 
-          unless record
-            record = emails.build(email: formatted_email)
-            valid_emails << record
+          if record.try(:confirmed?) || primary_email_record.nil? || options[:force_primary]
```

I think we should raise an exception if the email is not confirmed.  Silently ignore the unconfirmed emails might lead to weird bugs. 

And how about name the option `force` ?  `force_primary` is a bit duplicated and I am confused also. 

what you think ?